### PR TITLE
Serial Read and Write Node

### PR DESCRIPTION
**Pull request by noahzemlin**
Creates an `igvc_serial` package with a single node `serial_node`. This node can both read serial messages from serial devices and write to them. This pull request also includes the motor nucleo firmware. Finally, it updates the udev rules to name devices based on their serial number to ensure uniqueness.

### CODE REVIEW

File: firmware/motor_driver/IGVCMotor.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#ifndef IGVCMOTORH
+#define IGVCMOTORH
+
+class IGVCMotor {
+    public:
+        IGVCMotor(PinName enable, PinName pwm) {
```
Comment (jkleiber) [inline firmware/motor_driver/IGVCMotor.h:6]: Is there any reason we are implementing functions in the header file?

File: igvc_ws/src/igvc_serial/package.xml [hunk 2]
```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0"?>
+<package format="2">
+  <name>igvc_serial</name>
+  <version>0.0.0</version>
+  <description>The igvc_serial package</description>
+
+  <!-- One maintainer tag required, multiple allowed, one person per tag -->
+  <!-- Example:  -->
+  <!-- <maintainer email="jane.doe@example.com">Jane Doe</maintainer> -->
+  <maintainer email="kichang@todo.todo">kichang</maintainer>
```
Comment (jkleiber) [inline igvc_ws/src/igvc_serial/package.xml]: Not an issue, but would be good to figure out how we want to do maintainers and licenses

File: igvc_ws/src/igvc_serial/src/serial_node.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+class SerialReadThread(threading.Thread):
+    def __init__(self, serial_obj, topic):
+        threading.Thread.__init__(self)
+
+        self.serial_obj = serial_obj
+        self.publisher = rospy.Publisher(topic, String, queue_size=10)
+
+    def run(self):
+        # TODO: Is it good to use rospy.Rate here? Or should we just add a timeout
```
Comment (jkleiber) [inline igvc_ws/src/igvc_serial/src/serial_node.py]: As discussed, I think rospy.Rate will be fine here. That way the node's loop is controlled. The serial library should be able to handle this, and ROS will be happy

**Review by jkleiber — COMMENTED** [review]
Reviewed, please see comments

File: firmware/motor_driver/IGVCMotor.h (hunk 1 above)
Comment (noahzemlin) [reply firmware/motor_driver/IGVCMotor.h:6]: For such a short class it seems unnecessary to make two files. It also allows a quick view of all the functionality at once, if needed.

File: igvc_ws/src/igvc_serial/package.xml (hunk 2 above)
Comment (noahzemlin) [reply igvc_ws/src/igvc_serial/package.xml]: We should probably just have a maintainer for whoever worked on the package so it is easy to know who to ask questions about the package too. The license should just be MIT project-wide. I updated the project.xml to match this

File: igvc_ws/src/igvc_serial/src/serial_node.py (hunk 3 above)
Comment (noahzemlin) [reply igvc_ws/src/igvc_serial/src/serial_node.py]: Discussed.

**Pull request merged**